src/backend/distributed/utils/maintenanced.c

```diff
@@ -1040,12 +1040,30 @@ MaintenanceDaemonShmemExit(int code, Datum arg)
 	if (myDbData != NULL)
 	{
 		/*
-		 * Confirm that I am still the registered maintenance daemon before exiting.
+		 * Once the maintenance daemon fails (e.g., due to an error in the main loop),
+		 * both Postgres tries to restart the failed daemon and Citus attempt to start
```

Comment:
Are we sure Postgres restarts the maintenance deamon ever?
We are using [BGW_NEVER_RESTART](https://www.postgresql.org/docs/current/bgworker.html#:~:text=bgw_restart_time%20is%20the%20interval%2C%20in%20seconds%2C%20that%20postgres%20should%20wait%20before%20restarting%20the%20process%20in%20the%20event%20that%20it%20crashes.%20It%20can%20be%20any%20positive%20value%2C%20or%20BGW_NEVER_RESTART%2C%20indicating%20not%20to%20restart%20the%20process%20in%20case%20of%20a%20crash.) flag to prevent that.